src/ServiceControl.Monitoring/Metrics/Raw/OccurrenceSerializerDefinition.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace ServiceControl.Monitoring.Metrics.Raw
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using NServiceBus.MessageInterfaces;
+    using NServiceBus.Serialization;
+    using NServiceBus.Settings;
+
+    public class OccurrenceSerializerDefinition : SerializationDefinition
+    {
+        public override Func<IMessageMapper, IMessageSerializer> Configure(ReadOnlySettings settings)
+        {
+            return mapper => new OccurrenceSerializer();
+        }
+    }
+
+    public class OccurrenceSerializer : RawMessageSerializer<Occurrences>, IMessageSerializer
+    {
+        public object[] Deserialize(Stream stream, IList<Type> messageTypes = null)
+        {
+            return DeserializeRawMessage(stream, (t, r, m) => m.TryRecord(t));
```

Comment:
Is there a lambda allocation in here? If yes, could we extract the creation of the delegate and pass an already constructed one?